src/utils/__tests__/get-chunk-type-of-node.spec.ts

```diff
@@ -0,0 +1,44 @@
+import { chunkTypeOther, chunkTypeUnsortable } from "../../constants";
+import { getChunkTypeOfNode } from "../get-chunk-type-of-node";
+import { getImportNodes } from "../get-import-nodes";
+
+test('it classifies a default import as other', () => {
+    const importNodes = getImportNodes(`import a from "a";`);
+    expect(importNodes.length).toBe(1);
+    expect(getChunkTypeOfNode(importNodes[0])).toBe(chunkTypeOther);
+});
+
+test('it classifies a named import as other', () => {
+    const importNodes = getImportNodes(`import {a} from "a";`);
+    expect(importNodes.length).toBe(1);
+    expect(getChunkTypeOfNode(importNodes[0])).toBe(chunkTypeOther);
+});
+
+test('it classifies a side-effect import as unsortable', () => {
+    const importNodes = getImportNodes(`import "a";`);
+    expect(importNodes.length).toBe(1);
+    expect(getChunkTypeOfNode(importNodes[0])).toBe(chunkTypeUnsortable);
+});
+
+test('it classifies a named import with a ignore next line comment as unsortable', () => {
+    const importNodes = getImportNodes(`// prettier-ignore
+    import {a} from "a";`);
+    expect(importNodes.length).toBe(1);
+    expect(getChunkTypeOfNode(importNodes[0])).toBe(chunkTypeUnsortable);
+});
+
+test('it classifies a side-effect with a ignore next line comment as unsortable', () => {
+    const importNodes = getImportNodes(`// prettier-ignore
+    import "a";`);
+    expect(importNodes.length).toBe(1);
+    expect(getChunkTypeOfNode(importNodes[0])).toBe(chunkTypeUnsortable);
+});
+
+test('it only applies the ignore next line comments to the next line', () => {
+    const importNodes = getImportNodes(`// prettier-ignore
+    import {b} from "b";
+    import {a} from "a";`);
+    expect(importNodes.length).toBe(2);
+    expect(getChunkTypeOfNode(importNodes[0])).toBe(chunkTypeUnsortable);
+    expect(getChunkTypeOfNode(importNodes[1])).toBe(chunkTypeOther);
+});
```

Comment:
Nice-to-have:

Do you mind adding some tests for type imports, including the new typescript 4.5 syntax: `import {type Foo} from './foo';`?  From reading the code they should work fine, but might be nice to have explicit tests.

Reply:
I was half-way done with implementing ranged comments when I realized prettier doesn't actually support that for JavaScript. We can change the name, or I'd also not be opposed to removing support for ranged comments altogether. Normal single comments can do the job as well, and there's probably something else wrong when people need to ignore half of the imports.

As for type imports, yeah, shouldn't affect it, but I can add some tests.

Reply:
I'll leave the decision up to you.  Personally I think that you're right, ranged comments might not be needed.  It's less convenient to need a comment above every input, but this should be a rare escape-hatch anyway, it avoids some extra complexity, is easier to explain, and probably improves performance a tiny bit, if we don't support ranges.

Reply:
Ok, then I'll remove it. It makes the code more complex too and harder to understand.

For later, I was thinking about adding an option `unsortableImports` that let's users specify a pattern and if an import matches it, it is considered unsortable as well. This might be required for some non-standard usages of ES module imports that were mentioned in the issues over at trivago/prettier-plugin-sort-imports. If we do that, that's probably more helpful in the long term and doesn't require manual comments.

Reply:
Sounds good, let's see how far we can get with simple ignore statements, and if there's a demand we can add more features.